test/units/architecture/architect.test.js

```diff
@@ -1,22 +1,34 @@
 let { assert, expect } = require('chai');
 let should = require('chai').should();
-let carrot = require('../../../src/carrot');
+let {architect, Network, Node } = require('../../../src/carrot');
 
 /**
  *
  * There are 5 questions every unit test must answer.
  *
- * What is the unit under test (module, function, class, whatever)?
+ * What is the unit under test (module,fu nction, class, whatever)?
  * What should it do? (Prose description)
  * What was the actual output?
  * What was the expected output?
  * How do you reproduce the failure?
  *
  */
-it('should return true if valid user id', function(){
+describe('architect', function(){
+  describe('Perceptron', function() {
+    it('given input: 0, hidden: 0, output: 0, should throw Error', function(){
+           assert.throws(function() {  return new architect.Perceptron(0, 0, 0)  }, RangeError);
+       });
   
-});
+       it('should return false if invalid user id', function(){
```

Comment:
What is _"user id"_? 

Was this ported from StackOverflow? 

Did we add a `user_id` to `Network`?

Reply:
No, the first two tests was a walkthrough with Chris and I

Reply:
Got you! 

Let's not have those bleed through to production code.